runtime-testsuite/test/org/antlr/v4/test/runtime/go/BaseTest.java

```diff
@@ -129,36 +132,96 @@ protected void succeeded(Description description) {
 	@org.junit.Rule
 	public final Timeout eachTimeout = new Timeout(60000);
 
+	/**
+	 * Copies all files from go runtime to a temporary folder that is inside a valid GOPATH project structure.
+	 */
+	@BeforeClass
+	public static void setupGoRuntime() throws Exception {
+		tmpGopath = new File(System.getProperty("java.io.tmpdir"), "antlr-goruntime-tmpgopath-"
+				+ Long.toHexString(System.currentTimeMillis()));
+
+		ArrayList<String> pathsegments = new ArrayList<String>();
+		pathsegments.add("src");
+		pathsegments.addAll(Arrays.asList(GO_RUNTIME_IMPORT_PATH.split("/")));
+
+		File tmpPackageDir = tmpGopath;
+		for (String pathsegment : pathsegments) {
+			tmpPackageDir = new File(tmpPackageDir, pathsegment);
+		}
+		if (!tmpPackageDir.mkdirs()) {
+			throw new Exception("Could not create temp go runtime package dirs!");
+		}
+
+		File[] runtimeFiles = locateRuntime().listFiles();
+		if (runtimeFiles == null) {
+			throw new Exception("Go runtime file list is empty.");
+		}
+		for (File runtimeFile : runtimeFiles) {
+			File dest = new File(tmpPackageDir, runtimeFile.getName());
+			copyFile(runtimeFile, dest);
+		}
+
+		cacheGoRuntime(tmpPackageDir);
+	}
+
+	@AfterClass
+	public static void teardownGoRuntime() {
+		eraseDirectory(tmpGopath);
+	}
+
+	private static void cacheGoRuntime(File tmpPackageDir) throws Exception {
+		ProcessBuilder pb = new ProcessBuilder("go", "install", "-x");
+		pb.directory(tmpPackageDir);
+		pb.environment().put("GOPATH", tmpGopath.getPath());
+		pb.redirectErrorStream(true);
+		Process process = pb.start();
+		StreamVacuum sucker = new StreamVacuum(process.getInputStream());
+		sucker.start();
+		int exit = process.waitFor();
+		sucker.join();
+		if (exit != 0) {
+			throw new Exception("Non-zero exit while caching go runtime, output: " + sucker.toString());
+		}
+	}
+
+	private static void copyFile(File source, File dest) throws IOException {
+		InputStream is = new FileInputStream(source);
+		OutputStream os = new FileOutputStream(dest);
+		byte[] buf = new byte[4 << 10];
+		int l;
+		while ((l = is.read(buf)) > -1) {
+			os.write(buf, 0, l);
+		}
+		is.close();
+		os.close();
+	}
+
 	@Before
 	public void setUp() throws Exception {
 		// new output dir for each test
 		String prop = System.getProperty("antlr-go-test-dir");
 		if (prop != null && prop.length() > 0)
-			tmpdir = prop;
+			tmpdir = new File(prop);
 		else
-			tmpdir = new File(System.getProperty("java.io.tmpdir"), getClass()
-					.getSimpleName() + "-" + System.currentTimeMillis())
-					.getAbsolutePath();
+			tmpdir = new File(System.getProperty("java.io.tmpdir"),
+					getClass().getSimpleName() + "-" + System.currentTimeMillis());
```

Comment:
OK. That doesn't sound like it's a big deal to me either. The discrepancy raised a red flag. I guess it might be confusing if someone is poking around in this generated stuff and the timestamp is in two different formats.